src/Microsoft.TestPlatform.CrossPlatEngine/Client/ProxyDiscoveryManager.cs

```diff
@@ -294,5 +308,157 @@ private void LogMessage(TestMessageLevel testMessageLevel, string message)
             // Log to vstest.console layer.
             this.HandleLogMessage(testMessageLevel, message);
         }
+
+        private class ReflectionBasedRemoteTestDiscoveryEventHandler : ITestDiscoveryEventsHandler2
+        {
+            private ITestDiscoveryEventsHandler2 eventHandler;
+            private readonly IDataSerializer dataSerializer;
+
+            public ReflectionBasedRemoteTestDiscoveryEventHandler(ITestDiscoveryEventsHandler2 eventHandler, IDataSerializer dataSerializer)
+            {
+                this.eventHandler = eventHandler;
+                this.dataSerializer = dataSerializer;
+            }
+
+            public void HandleDiscoveredTests(IEnumerable<TestCase> discoveredTestCases)
+            {
```

Comment:
Here we get Source that is the actual remote path, and we need to translate it back to the appxrecipe we got originally. So TestWindow can match it back to the correct source.